app/main.py

```
@@ -130,32 +130,68 @@ async def health_check(response: Response) -> HealthResponse:
-        logger.error(f"Health check timeout (>{HEALTH_CHECK_TIMEOUT}s): Supabase connection took too long", exc_info=True)
+        logger.error(f"Health check timeout (>{HEALTH_CHECK_TIMEOUT}s): Supabase DB query took too long.", exc_info=True)
         health_status["status"] = "degraded"
         health_status["dependencies"]["database"] = "down"
         response.status_code = 503
-        return HealthResponse(**health_status)
-    except Exception:  # noqa: BLE001
-        logger.error("Health check failed: Supabase connection error", exc_info=True)
+    except httpx.HTTPStatusError as e:
+        status_code = getattr(e.response, 'status_code', None)
+        if status_code == 401:
+            logger.error(f"Health check failed with HTTP 401: Invalid API key. Check SUPABASE_KEY.", exc_info=True)
```

Ruff (0.15.2)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)